src/hubbleds/utils.py

```diff
@@ -126,3 +133,61 @@ def data_summary_for_component(data, component_id):
         summary[f"{percent}%"] = (bottom, top)
 
     return summary
+
+
+M = TypeVar("M", bound=BaseModel)
+def models_to_glue_data(items: List[M],
+                        label: str | None=None,
+                        ignore_components: list[str] | None=None
+) -> Data:
+    data_dict = {}
+    if items:
+        t = type(items[0])
+        ignore = ignore_components or []
+        for field in t.model_fields.keys():
+            if field not in ignore:
+                data_dict[field] = [getattr(m, field) for m in items]
+    if label:
+        data_dict["label"] = label
+    return Data(**data_dict)
```

Comment:
This seems a bit more lax that cosmicds/cosmicds#303, do we risk creating `Component` vs `CategoricalComponent` elements in the `Data` object?

Reply:
Good call - I'll update this to use the same logic as in the cosmicds PR (actually, I'll probably refactor the component-choosing logic into its own function that we can reuse here).